tests/test_iterator.rs

```diff
@@ -223,39 +223,51 @@ fn test_total_order_seek() {
     let mut opts = Options::new();
     opts.create_if_missing(true);
     opts.set_block_based_table_factory(&bbto);
-    opts.set_prefix_extractor("FixedSuffixTransform",
-                              Box::new(FixedSuffixTransform { suffix_len: 2 }))
+    opts.set_prefix_extractor("FixedPrefixTransform",
+                              Box::new(FixedPrefixTransform { prefix_len: 2 }))
         .unwrap();
     // also create prefix bloom for memtable
     opts.set_memtable_prefix_bloom_size_ratio(0.1 as f64);
 
-    let keys = vec![b"k1-0", b"k1-1", b"k1-2", b"k2-0", b"k2-1", b"k2-2", b"k3-0", b"k3-1",
-                    b"k3-2"];
+    let keys = vec![b"k1-1", b"k1-2", b"k1-3", b"k2-1", b"k2-2", b"k2-3", b"k3-1", b"k3-2",
+                    b"k3-3"];
     let db = DB::open(opts, path.path().to_str().unwrap()).unwrap();
     let wopts = WriteOptions::new();
 
     // sst1
-    db.put_opt(b"k1-0", b"a", &wopts).unwrap();
-    db.put_opt(b"k1-1", b"b", &wopts).unwrap();
-    db.put_opt(b"k1-2", b"c", &wopts).unwrap();
+    db.put_opt(b"k1-1", b"a", &wopts).unwrap();
+    db.put_opt(b"k1-2", b"b", &wopts).unwrap();
+    db.put_opt(b"k1-3", b"c", &wopts).unwrap();
+    db.put_opt(b"k2-1", b"a", &wopts).unwrap();
     db.flush(true /* sync */).unwrap(); // flush memtable to sst file.
 
     // sst2
-    db.put_opt(b"k2-0", b"a", &wopts).unwrap();
-    db.put_opt(b"k2-1", b"b", &wopts).unwrap();
-    db.put_opt(b"k2-2", b"c", &wopts).unwrap();
+    db.put_opt(b"k2-2", b"b", &wopts).unwrap();
+    db.put_opt(b"k2-3", b"c", &wopts).unwrap();
     db.flush(true /* sync */).unwrap(); // flush memtable to sst file.
 
     // memtable
-    db.put_opt(b"k3-0", b"a", &wopts).unwrap();
-    db.put_opt(b"k3-1", b"b", &wopts).unwrap();
-    db.put_opt(b"k3-2", b"c", &wopts).unwrap();
+    db.put_opt(b"k3-1", b"a", &wopts).unwrap();
+    db.put_opt(b"k3-2", b"b", &wopts).unwrap();
+    db.put_opt(b"k3-3", b"c", &wopts).unwrap();
 
     let mut iter = db.iter();
     iter.seek(SeekKey::Key(b"k1-0"));
     let mut key_count = 0;
     while iter.valid() {
-        // only iterator sst files and memtable that contain keys has the same prefix with b"k1-0".
+        assert_eq!(keys[key_count], iter.key());
+        key_count = key_count + 1;
+        iter.next();
+    }
+    assert!(key_count == 4);
+
+    let mut ropts = ReadOptions::new();
+    ropts.set_prefix_same_as_start(true);
+    let mut iter = db.iter_opt(ropts);
+    iter.seek(SeekKey::Key(b"k1-0"));
+    let mut key_count = 0;
+    while iter.valid() {
+        // only iterator sst files and memtable that contain keys has the same prefix with b"k1"
```

Comment:
and become invalid as soon as encounter a key with prefix not the same as start key.